lib/builder.js

```diff
@@ -120,12 +110,45 @@ exports.writeOutputFile = function(opts, outputs, baseURL) {
 
   return asp(mkdirp)(path.dirname(path.resolve(basePath, opts.outFile)))
   .then(function() {
-    if (!opts.sourceMaps)
-      return;
-    return asp(fs.writeFile)(path.resolve(basePath, path.dirname(opts.outFile), sourceMapFile), output.sourceMap);
+    if (!opts.sourceMaps) return;
+    var sourceMapPath = path.resolve(basePath, path.dirname(opts.outFile), sourceMapFile);
+    return asp(fs.writeFile)(sourceMapPath, output.sourceMap);
   })
   .then(function() {
-    return asp(fs.writeFile)(path.resolve(basePath, opts.outFile), output.source);
+    var sourcePath = path.resolve(basePath, opts.outFile);
+    return asp(fs.writeFile)(sourcePath, output.source);
+  });
+}
+
+function wrapPromise(value) {
+  return new rsvp.Promise(function(resolve, reject) {
+    return resolve(value);
   });
+}
+
+function writeOutputString(opts, output, baseURL) {
+  var content = output.source;
+  if (opts.sourceMaps) {
+    content += '\n//# sourceMappingURL=data:application/json;base64,' +
+      new Buffer(output.sourceMap.toString()).toString('base64');
```

Comment:
I think even with a memory option output, it's fine to return the source map as a separate return value rather than encoding, as that's easier for chaining in custom builds.